docs/fabric_mod_json_spec.md

```diff
@@ -0,0 +1,130 @@
+# fabric.mod.json
+
+In all cases, the mod JSON, `fabric.mod.json`, is defined as either an object containing a "schemaVersion" key with an integer value, denoting the version of the format.
+
+If a "schemaVersion" key is missing, assume version "0".
+
+It is important to remember that the Fabric mod JSON is **authoritative**. This means that Loader relies on it specifically to gather the necessary information - from the perspective of an external tool, its contents can be trusted and relied upon.
+
```

Comment:
We should probably specify the behavior with regards to duplicate k->v values in the FMJ.

The way this is right now is we use the latter field EXCEPT for the `schemaVersion` which we fail to parse when there are multiple instances of it.